src/phylo/nodes.jl

```diff
@@ -0,0 +1,655 @@
+
+**Parameters:**
+
+* `x`:    The PhyNode to set the confidence of.
+
+* `conf`: The value of the confidence to be set.
+""" ->
+function confidence!(x::PhyNode, conf::Float64)
+  x.confidence = conf
+end
+
+@Docile.doc """
+Set the confidence of the node.
+
+**Parameters:**
+
+* `x`:    The PhyNode to set the confidence of.
+
+* `conf`: The value of the confidence to be set.
+
+
+""" ->
+function confidence!(x::PhyNode, conf::Nothing)
+  x.confidence = -1.0
+end
+
+@Docile.doc """
+Test whether a node is empty.
+
+**Parameters:**
+
+* `x`: The PhyNode to test.
+""" ->
+function isempty(x::PhyNode)
+  return x.name == "" && !blisknown(x) && !hasextensions(x) && !haschildren(x) && parentisself(x) && !confisknown(x)
+end
+
+@Docile.doc """
+Get the name of a PhyNode.
+
+**Parameters:**
+
+* `x`: The PhyNode to get the name of.
+""" ->
+function name(x::PhyNode)
+  return x.name
+end
+
+@Docile.doc """
+Get the branch length of a PhyNode.
+
+**Parameters:**
+
+* `x`: The PhyNode to get the branch length of.
+""" ->
+function branchlength(x::PhyNode)
+  return x.branchlength
+end
+
+@Docile.doc """
+Test whether a node is a leaf.
+
+**Parameters:**
+
+* `x`: The PhyNode to test.
+""" ->
+function isleaf(x::PhyNode)
+  return hasparent(x) && !haschildren(x)
+end
+
+@Docile.doc """
+Test whether a node has children.
+
+**Parameters:**
+
+* `x`: The PhyNode to test.
+""" ->
+function haschildren(x::PhyNode)
+  return length(x.children) > 0
+end
+
+@Docile.doc """
+Test whether a node is the parent of another specific node.
+
+**Parameters:**
+
+* `parent`: The potential parent PhyNode to test.
+
+* `child`:  The potential child PhyNode to test.
+""" ->
+function haschild(parent::PhyNode, child::PhyNode)
+  return in(child, parent.children)
+end
+
+@Docile.doc """
+Test whether a node is its own parent. See PhyNode().
+
+**Parameters:**
+
+* `x`: The PhyNode to test.
+""" ->
+function parentisself(x::PhyNode)
+  return x.parent === x
+end
+
+@Docile.doc """
+Test whether a node has a parent.
+
+**Parameters:**
+
+* `x`: The PhyNode to test.
+""" ->
+function hasparent(x::PhyNode)
+  return !parentisself(x)
+end
+
+@Docile.doc """
+Get the children of a node.
+
+**Parameters:**
+
+* `x`: The PhyNode to get children for.
+""" ->
+function children(x::PhyNode)
+  return x.children
+end
+
+@Docile.doc """
+Get the siblings of a node. Included in output is the input node.
+
+**Parameters:**
+
+* `x`: The PhyNode to get siblings for.
+""" ->
+function siblings(x::PhyNode)
+  if hasparent(x)
+    return children(x.parent)
+  end
+end
+
+@Docile.doc """
+Get the parent of a node.
+
+**Parameters:**
+
+* `x`: The PhyNode to get the parent of.
+""" ->
+function parent(x::PhyNode)
+  if parentisself(x)
+    println("Node does not have a parent. It is self referential.")
+  end
+  return x.parent
+end
+
+@Docile.doc """
+Test whether a node is the root node.
+
+**Parameters:**
+
+* `x`: The PhyNode to test.
+""" ->
+function isroot(x::PhyNode)
```

Comment:
That's something we need to decide. Because if a node has no children, and no parent, then it is not part of any phylogeny. 

It is a single, unconnected node and cannot be called a Phylogeny in any meaningful way: Even the most meaningless phylogenetic structure (totally unresolved phylogenies) have a central root node with every tip (every organism or species) branching off it. This is (called an unresolved star phylogeny). If that is as simple as phylogenies can be, then a single unconnected node should not be called one, and so isn't a root insofar as it isn't even part of a phylogeny of which to be a root. This is why they instead of returning true for isroot, they return false. But I believe there is another method that checks is a node is part of a tree / is a single unconnected node.